var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -290,6 +290,10 @@ def libs(self):
     # and https://github.com/boostorg/context/issues/177
     patch("context-macho-gcc.patch", when="@1.65:1.76 +context platform=darwin %gcc")
 
+    # Fix float128 support when building with CUDA and Cray compiler
+    # See https://github.com/boostorg/config/pull/378
+    patch('fee1ad07968386b6d547f089311b7a2c1bf7fa55.patch', when='@:1.76%cce')
```

Comment:
```suggestion
    patch('https://github.com/boostorg/config/commit/fee1ad07968386b6d547f089311b7a2c1bf7fa55.patch', when='@:1.76%cce')
```

Reply:
I tried this first, but did not get the patch to apply. Essentially, the path is different. In the `boostorg/config` patch the path is `include/boost/config/detail/suffix.hpp` and I changed it to `boost/config/detail/suffix.hpp` in the patch file in this PR. I don't know if there's a way to patch the patch... I could also not find a way to tell the spack `patch` function a relative directory/file to apply it to. Did I miss it?

Reply:
This worked for me:
```suggestion
    patch(
        'https://github.com/boostorg/config/commit/fee1ad07968386b6d547f089311b7a2c1bf7fa55.patch',
        sha256='3b159d65a0d3d2df2a21c6bf56ffaba943fce92d2d41d628b2c4d2e924e0f421',
        when='@:1.76', level=2
   )
```

Reply:
Nice, I didn't realize that existed. Thanks! I've pushed this change and removed the patch file.